### DISCUSSION TITLE
Produce an ESM build/change usage of require('buffer')?

### DISCUSSION THREAD

**Discussion by danawoodman** [alexandercerutti/passkit-generator#117] [Ideas]
I'm attempting to generate an ESM build that I'm going to use in a CloudFlare Worker. 

The current examples point to using Webpack, which I would very much like to avoid for a variety of reasons, especially dev/build speed. 

I've had success using tsup for bundling for workers in other projects, but using passkit-generator fails with:

```
Error: Dynamic require of "buffer" is not supported
    at file:///Users/danawoodman/code/wallet/dist/index.mjs:12:9
    at node_modules/passkit-generator/lib/PKPass.js (file:///Users/danawoodman/code/wallet/dist/index.mjs:29251:20)
    at __require2 (file:///Users/danawoodman/code/wallet/dist/index.mjs:18:50)
    at node_modules/passkit-generator/lib/index.js (file:///Users/danawoodman/code/wallet/dist/index.mjs:29653:20)
    at __require2 (file:///Users/danawoodman/code/wallet/dist/index.mjs:18:50)
    at file:///Users/danawoodman/code/wallet/dist/index.mjs:29664:40
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:527:24)
    at async loadESM (node:internal/process/esm_loader:91:5)

Node.js v18.9.0
```

If I change to a cjs export then I get the following errors:

```
▲ [WARNING] Converting "require" to "esm" is currently not supported [unsupported-require-call]

    dist/index.js:29305:19:
      29305 │     var buffer_1 = require("buffer");
```

I've tried many workarounds but they're all failing at the usage of Buffer in this project and I'm not sure where to go from here.

Is there plans to generate an ESM build and/or change how buffer is required in the project to not require the usage of import rewriting like is required when using Webpack?

**Comment by alexandercerutti** [comment]
Hey @danawoodman, current examples point to Webpack because they actually offer a way to build which involves Webpack. Otherwise, I set up an example with Webpack 5 as an external tool. So, this does not actually preclude the possibility to use a different bundler. I never used `tsup`, but I might try.

I think the issue here is that Cloudflare Workers do not actually have support for Buffers, hence it should be polyfilled. I don't know how this is possible with `tsup`.

Let me know if by doing this you are able to solve.
